[expo-media-library] migrateAlbumIfNeededAsync causes IllegalArgumentException since expo 48

### Minimal reproducible example

https://github.com/a-zen/expo-album-migrate-error

### Summary

After the upgrade to expo 48, migrating an album fails with the following error message on Android with Expo GO App:

` ERROR  Sat, 01 Apr 2023 14:39:41 GMT ERROR: addPhotoToCameraRoll: Error: Call to function 'ExpoMediaLibrary.migrateAlbumIfNeededAsync' has been rejected.
→ Caused by: java.lang.IllegalArgumentException: the bind value at index 1 is null`

when exeucting the code

````
import * as MediaLibrary from 'expo-media-library';
// ....
await MediaLibrary.migrateAlbumIfNeededAsync(album);
````

like described in the documentation [here](https://docs.expo.dev/versions/v48.0.0/sdk/media-library/#medialibrarymigratealbumifneededasyncalbum).

When reverting back to expo 47 it works like before.

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 6.2 Arch Linux
      Shell: 5.1.16 - /nix/store/kqbccy5vkrnbx0jb2klzx1sl4cjxpvzh-bash-interactive-5.1-p16/bin/bash
    Binaries:
      Node: 14.20.0 - /nix/store/4cmjll22h0gdm2m94nl4wf1kgrl7bc3v-nodejs-slim-14.20.0/bin/node
      Yarn: 1.22.19 - /nix/store/slbgjyqgadyzgbihc8k7mzl635ddgjhf-yarn-1.22.19/bin/yarn
      npm: 8.12.2 - /nix/store/ncwvnf72gcyk565ma54a3fpicdg3czam-npm-8.12.2/bin/npm
    SDKs:
      Android SDK:
        Build Tools: 34.0.0
        System Images: android-31 | Google APIs Intel x86 Atom_64
    IDEs:
      Android Studio: AI-221.6008.13.2211.9619390
    npmPackages:
      babel-preset-expo: ^9.3.0 => 9.3.0 
      expo: 48.0.9 => 48.0.9 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.71.4 => 0.71.4 
      react-native-web: ~0.18.7 => 0.18.12 
    Expo Workflow: managed

I just tested this with Expo 49 again and the error is still there.

````WARN  Possible Unhandled Promise Rejection (id: 0):
Error: Call to function 'ExpoMediaLibrary.migrateAlbumIfNeededAsync' has been rejected.
→ Caused by: java.lang.IllegalArgumentException: the bind value at index 1 is null
Error: Call to function 'ExpoMediaLibrary.migrateAlbumIfNeededAsync' has been rejected.
→ Caused by: java.lang.IllegalArgumentException: the bind value at index 1 is null
    at construct (native)
    at apply (native)
    at _construct (http://192.168.2.40:8081/node_modules/expo/AppEntry.bundle//&platform=android&dev=true&hot=false&lazy=true:4486:28)
    at Wrapper (http://192.168.2.40:8081/node_modules/expo/AppEntry.bundle//&platform=android&dev=true&hot=false&lazy=true:4448:25)
    at construct (native)
    at _createSuperInternal (http://192.168.2.40:8081/node_modules/expo/AppEntry.bundle//&platform=android&dev=true&hot=false&lazy=true:125316:322)
    at call (native)
    at CodedError (http://192.168.2.40:8081/node_modules/expo/AppEntry.bundle//&platform=android&dev=true&hot=false&lazy=true:125329:26)
````

````
  expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 6.3 Arch Linux
      Shell: 5.2.15 - /nix/store/148x9bca5p18var9kj79pc3d589djrk8-bash-interactive-5.2-p15/bin/bash
    Binaries:
      Node: 20.3.1 - /nix/store/vlf4b5nm3q0ykbv97i99gvwpw3k8dphr-nodejs-20.3.1/bin/node
      npm: 9.7.2 - /nix/store/aj8jqnqkx5k4b4a1q2rndr30vfsi209i-npm-9.7.2/bin/npm
    SDKs:
      Android SDK:
        Build Tools: 34.0.0
        System Images: android-31 | Google APIs Intel x86 Atom_64
    IDEs:
      Android Studio: AI-222.4459.24.2221.10121639
    npmPackages:
      expo: ^49.0.0 => 49.0.0 
      react: 18.2.0 => 18.2.0 
      react-native: 0.72.1 => 0.72.1 
    Expo Workflow: managed

````

--- a/packages/expo-media-library/CHANGELOG.md
+++ b/packages/expo-media-library/CHANGELOG.md
@@ -8,7 +8,7 @@
 
 ### 🐛 Bug fixes
 
-- Fix `the bind value at index 1 is null` exception thrown by `ExpoMediaLibrary.migrateAlbumIfNeededAsync` on Android.
+- Fix `the bind value at index 1 is null` exception thrown by `ExpoMediaLibrary.migrateAlbumIfNeededAsync` on Android. ([#23515](https://github.com/expo/expo/pull/23515) by [@lukmccall](https://github.com/lukmccall))
 
 ### 💡 Others
 

--- a/packages/expo-media-library/CHANGELOG.md
+++ b/packages/expo-media-library/CHANGELOG.md
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- Fix `the bind value at index 1 is null` exception thrown by `ExpoMediaLibrary.migrateAlbumIfNeededAsync` on Android.
+
 ### 💡 Others
 
 ## 15.4.0 — 2023-06-21

--- a/packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryModule.kt
+++ b/packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryModule.kt
@@ -218,10 +218,18 @@ class MediaLibraryModule : Module() {
         return@AsyncFunction
       }
 
+      val assetsIds = getAssetsInAlbums(context, albumId)
+        .filter { it.isNotEmpty() }
+        .toTypedArray()
+      // The album is empty, nothing to migrate
+      if (assetsIds.isEmpty()) {
+        return@AsyncFunction
+      }
+
       val assets = MediaLibraryUtils.getAssetsById(
         context,
         null,
-        *getAssetsInAlbums(context, albumId).toTypedArray()
+        *assetsIds
       )
 
       val albumsMap = assets

